src/+otp/+kortewegdevries/f.m

```diff
@@ -0,0 +1,6 @@
+function du = f(t, u, Theta, Alpha, Nu, Rho, Dx, D3x)
+
+u_x = Dx*u;
+du = Alpha*(Theta*(Dx*(u.^2)) + 2*(1 - Theta)*(u.*u_x)) + Rho*u_x + Nu*(D3x*u);
```

Comment:
We want to be able to write every problem in mass-matrix form, so the invariant should be part of f.

You can have a separate fdynamics and invariant functions that are part of an alternate RHS.

Reply:
This problem is never solved as a DAE. Its always solved as a PDE. So I dont think the invariant should be part of it.

Reply:
Is the problem solved using the invariant at all, or is the invariant preserved without the need to add it to the RHS?

Reply:
With the right time integration method, it is preserved without adding it to RHS. Implicit midpoint is amazing for this problem

Reply:
OK that's fine. I see. Then it's fine to do what you do now. No more objections.